### ISSUE TITLE
[sabnzbd develop branch] spotweb gives error when pushing to sabnzbd

### ISSUE THREAD

**Issue by Hydranet** [sabnzbd/sabnzbd#1501]
I am running the develop branch of sabnzbd, I just noticed today that when I login to spotweb and push an nzb to sabnzbd the green download error gets a red cross through it as if the pushing has failed. However sabnzbd does receive the nzb and it does start downloading, I tried it with other titles and I get the same result there.

I see this in the apache logs:
`[Thu Jun 11 19:56:33.655419 2020] [php7:notice] [pid 6121] [client 0.0.0.0.29:33658] sabnzbd push : http returncode: 200 /  / {'status': True, 'nzo_ids': ['SABnzbd_nzo_j4cmwk47']}\n, referer: https://spotweb.feedmebits.nl/?page=index&search[tree]=cat0_z0_a9&search[value][]=`

In my sabnzbd logs I see sabnzbd receiving the file from spotweb and starting to download:
[sabnzbd.txt](https://github.com/sabnzbd/sabnzbd/files/4766589/sabnzbd.txt)

Here is what I see in spotweb when I try it: 
![Screenshot from 2020-06-11 19-44-18](https://user-images.githubusercontent.com/22844936/84421726-467ed380-ac1c-11ea-99c9-a5ae2137e6e2.png)

The really strange thing is  when I add the download to sonarr/radarr and then request my searcher to look for it on my spotweb indexer I don't get any errors in de apache logs. When I then refresh spotweb it shows it correctly download the a green check mark.

![Screenshot from 2020-06-11 20-23-50](https://user-images.githubusercontent.com/22844936/84425273-90b68380-ac21-11ea-8064-b165ed526c49.png)

I am using: 
- python36 (3.6.8)
- CentOS8
- apache 2.4 (Running sabnzbd behind apache via a reverse proxy)

spotweb hasn't been updated in a few months:
```
commit bf7d95d18576707cfb4c1dc77181fca3fba7cc97
Author: Sweepr <Sweepr@users.noreply.github.com>
Date:   Sat Apr 4 22:10:48 2020 +0200

    Update README.md
```
So thinking it may have been one  of the recent updates to the develop branch.

I don't log into spotweb very often since I use radarr and sonarr so I wouldn't be able to tell you since when this started.

**Comment by sanderjo** [comment]
You have 79 Tracebacks in your log! Shouldn't you solve that first?

**Comment by Safihre** [comment]
I did make a change to the API, the output has changed slightly.
Maybe Spotnet can't handle that.
Can you see the request spotnet made for this response?
```
[Thu Jun 11 19:56:33.655419 2020] [php7:notice] [pid 6121] [client 0.0.0.0.29:33658] sabnzbd push : http returncode: 200 / / {'status': True, 'nzo_ids': ['SABnzbd_nzo_j4cmwk47']}\n, referer: https://spotweb.feedmebits.nl/?page=index&search[tree]=cat0_z0_a9&search[value][]=
```

**Comment by Hydranet** [comment]
@sanderjo Those were from yesterday, I solved those yesterday. The entries with today's date don't have any tracebacks, I did check those.  I check the downloads I did today via spotweb and there I didn't see any tracebacks, neither when I scorlled through all today's logs.

**Comment by Hydranet** [comment]
> I did make a change to the API, the output has changed slightly.
> Maybe Spotnet can't handle that. But I also haven't made a release yet since that change, so nobody could pick up on it.
> Can you see the request spotnet made for this response?
> 
> ```
> [Thu Jun 11 19:56:33.655419 2020] [php7:notice] [pid 6121] [client 0.0.0.0.29:33658] sabnzbd push : http returncode: 200 / / {'status': True, 'nzo_ids': ['SABnzbd_nzo_j4cmwk47']}\n, referer: https://spotweb.feedmebits.nl/?page=index&search[tree]=cat0_z0_a9&search[value][]=
> ```
@Safihre 
I tried that, when I open that url it just opens the category the movie is from which I tried downloading via spotweb.
![Screenshot from 2020-06-11 21-14-28](https://user-images.githubusercontent.com/22844936/84429517-94013d80-ac28-11ea-9275-149622548b24.png)

**Comment by Safihre** [comment]
The change I made is this: 
Long, long time ago, SABnzbd didn't support JSON output and all the API-calls would return plain text responses. When adding an NZB it would just respond `ok`. 
Around 2012 we added JSON output in which we respond with the ID's of the jobs that were added (since it can be multiple). For backwards-compatibility a special mode was added that could still output `ok`, this is what Spotnet uses.
After 8 more years, I decided to remove this backwards compatibility.

**Comment by Hydranet** [comment]
In which commit id did you remove that? I could try undoing that commit and then seeing how spotweb reacts before that commit in the develop branch.

**Comment by Safihre** [comment]
https://github.com/sabnzbd/sabnzbd/commit/308c0ec621026cf08b12dd761f7975112499d4a7

**Comment by Hydranet** [comment]
When I switch to the previous commit it works as expected

```
commit 2f45a87f125aa1bf6c2b595b767f3eb0c4b247e5
Author: Safihre <safihre@sabnzbd.org>
Date:   Mon Jun 8 16:57:18 2020 +0200

    API-method addurl only supports adding 1 URL at a time

```

**Comment by Safihre** [comment]
While annoying, I do want to use this major release to including breaking changes. Otherwise it might be another 8 years 🤔
Spotnet will have to adjust..

**Comment by thezoggy** [comment]
looks like spotnab sends `output=jsonp`  which is invalid. should be *json*. so all this time sab just silently handled that typo and now we dont

https://github.com/spotweb/spotweb/blob/develop/lib/services/NzbHandler/Services_NzbHandler_Clientsabnzbd.php#L14

**Comment by thezoggy** [comment]
and about the issue adding, spotnab uses output=text and then checks if 'ok' was returned
https://github.com/spotweb/spotweb/blob/develop/lib/services/NzbHandler/Services_NzbHandler_Pushsabnzbd.php#L56

they should just update to json and handle that.. as that will fix 3.x+ support and still work fine with 2.x and older as well

**Comment by Safihre** [comment]
Thanks for looking into that!

**Comment by Hydranet** [comment]
@thezoggy 

Thanks! I created an issue for spotweb referring back to this one.

**Comment by sanderjo** [comment]
> @thezoggy
> 
> Thanks! I created an issue for spotweb referring back to this one.

With the two advices of @thezoggy , maybe you can patching it yourself and if it works send a PR to spotweb?

**Comment by mesa57** [comment]
@Safihre : 
Could you also amend you're api documentation ? 

Request types

Supported output types are json, xml and text and can be specified in the request:

http://host:port/sabnzbd/api?output=json

**Comment by Safihre** [comment]
Text is still supported, but it doesn't just output `ok` anymore but more details. That's all that changed 🙂

**Comment by thezoggy** [comment]
as a side note, this change also breaks the original sickbeard (as it also checked for 'ok'). which is one of the reasons why shy left this the way it was for so long
https://github.com/midgetspy/Sick-Beard/blob/development/sickbeard/sab.py#L101

**Comment by sanderjo** [comment]
Thanks to analysis of @thezoggy I've reported a bug report at Spotweb.

**Comment by Sweepr** [comment]
This fix has now been released into version 1.4.6 as of yesterday.

See: https://github.com/spotweb/spotweb/tree/1.4.6
